### PR TITLE
Tracking of completeness of the implementation

### PR DESCRIPTION
**Pull request by sdasgup3**
fixes https://github.com/openxla/stablehlo/issues/16

### CODE REVIEW

File: docs/implementation_progress.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+tracked using one of the following tracking labels:
+
+ - yes: complete
+ - wip: semi-complete: Work in progress or under review.
+ - no: not complete yet, but plan to do so in near future.
+ - no-plan: not complete and not planned.
+
+## Audit chart
+
+| StableHLO Operation (114) | Specification () | Verifier | Shape-Inference | Pretty-Printer | Interpreter (2) |
```
Comment (zhouxin913) [inline docs/implementation_progress.md:25]: "Shape-Inference" -> "Type Inference" I would suggest.

File: docs/implementation_progress.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+still plenty to do to review the existing implementations for completeness and
+provide new implementations where none exist.
+
+This live document is for the developers to track the progress on various
+aspects of the implementation - pretty-printing, verification, shape inference,
+specification, interpreter etc.
+
+### How to use it
+
+The progress of a stableHLO op, as mentioned in the corresponding row, on a
```
Comment (zhouxin913) [inline docs/implementation_progress.md:14]: stableHLO->StableHLO ?

**Comment by sdasgup3** [comment]
Added the tracking labels to track the completeness of verifier based on its with with XLA and/or StableHLO semantics. Also, feel free to add customized labels (as shown in the doc for verifier) to better track your progress.

**Review by zhouxin913 — APPROVED** [review]

File: docs/implementation_progress.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
+## About
+
+When bootstrapping StableHLO from MHLO, we have inherited MHLO's implementation
+of verification. It already has significant coverage of the opset, but there's
```
Comment (burmako) [inline docs/implementation_progress.md:4]: "of many things, including prettyprinting, verification and shape inference". It's not just verification that we inherited, right?

File: docs/implementation_progress.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
+## About
```
Comment (burmako) [inline docs/implementation_progress.md:1]: Let's come up with a shorter name? E.g. `status.md` or `progress.md`.

File: docs/implementation_progress.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+When bootstrapping StableHLO from MHLO, we have inherited MHLO's implementation
+of verification. It already has significant coverage of the opset, but there's
+still plenty to do to review the existing implementations for completeness and
+provide new implementations where none exist.
+
+This live document is for the developers to track the progress on various
```
Comment (burmako) [inline docs/implementation_progress.md:8]: "for developers and users".

File: docs/implementation_progress.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    - **no**: not complete yet, but plan to do so in near future.
+    - **no-plan**: not complete and not planned.
+ - Customized labels
+    - Verifier
+       - **match-xla**:  verifier in sync with  [XLA semantics](https://www.tensorflow.org/xla/operation_semantics).
+       - **match-spec**: verifier in sync with [StableHLO semantics](https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md).
+
+## Audit chart
+
+| StableHLO Operation (114) | Specification (21) | Verifier | Type Inference | Pretty Printer | Interpreter (2) |
```
Comment (burmako) [inline docs/implementation_progress.md:30]: "Verification" and "Prettyprinting" to match labels?

File: docs/implementation_progress.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+### How to use it
+
+The progress of a StableHLO op, as mentioned in the corresponding row, on a
+particular implementation aspect, as mentioned in the corresponding column, is
+tracked using one of the following tracking labels.
+
+ - Generic labels
+    - **yes**: complete
+    - **wip**: semi-complete: Work in progress or under review.
+    - **no**: not complete yet, but plan to do so in near future.
```
Comment (burmako) [inline docs/implementation_progress.md:21]: How about "not complete yet, but part of [the roadmap](https://github.com/openxla/stablehlo#roadmap)"?

File: docs/implementation_progress.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    - **no-plan**: not complete and not planned.
```
Comment (burmako) [inline docs/implementation_progress.md:22]: Do we expect to use this label for something right away? If not, let's drop it. We can always bring it back if needed later.

File: docs/implementation_progress.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+ - Customized labels
+    - Verifier
+       - **match-xla**:  verifier in sync with  [XLA semantics](https://www.tensorflow.org/xla/operation_semantics).
+       - **match-spec**: verifier in sync with [StableHLO semantics](https://github.com/openxla/stablehlo/blob/main/docs/spec_draft.md).
+
+## Audit chart
```
Comment (burmako) [inline docs/implementation_progress.md:28]: Consider simplifying to "Status". "Audit" matches the internal spreadsheet, but that's mostly a historical name that wasn't part of opensource story.

**Review by burmako — COMMENTED** [review]
I'm a bit late to the party, so I provided feedback in a follow-up PR: https://github.com/openxla/stablehlo/pull/136.

**Pull request merged**